<a href="https://colab.research.google.com/github/Rahul081203/ML_projects/blob/master/Rock_V_s_Mine.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Importing The Dependencies for Project Initialization

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

#DATA COLLECTION

In [ ]:
data=pd.read_csv('/content/sonar all-data.csv',header=None) # Loading a pandas dataframe

In [ ]:
data.shape # shape of the dataset

(208, 61)

There are 60 Features and 1 target column with 208 examples

In [ ]:
data.describe() #stats

0           1           2           3           4           5   \
count  208.000000  208.000000  208.000000  208.000000  208.000000  208.000000   
mean     0.029164    0.038437    0.043832    0.053892    0.075202    0.104570   
std      0.022991    0.032960    0.038428    0.046528    0.055552    0.059105   
min      0.001500    0.000600    0.001500    0.005800    0.006700    0.010200   
25%      0.013350    0.016450    0.018950    0.024375    0.038050    0.067025   
50%      0.022800    0.030800    0.034300    0.044050    0.062500    0.092150   
75%      0.035550    0.047950    0.057950    0.064500    0.100275    0.134125   
max      0.137100    0.233900    0.305900    0.426400    0.401000    0.382300   

               6           7           8           9   ...          50  \
count  208.000000  208.000000  208.000000  208.000000  ...  208.000000   
mean     0.121747    0.134799    0.178003    0.208259  ...    0.016069   
std      0.061788    0.085152    0.118387    0.134416  ...    0.012008   
min      0.003300    0.005500    0.007500    0.011300  ...    0.000000   
25%      0.080900    0.080425    0.097025    0.111275  ...    0.008425   
50%      0.106950    0.112100    0.152250    0.182400  ...    0.013900   
75%      0.154000    0.169600    0.233425    0.268700  ...    0.020825   
max      0.372900    0.459000    0.682800    0.710600  ...    0.100400   

               51          52          53          54          55          56  \
count  208.000000  208.000000  208.000000  208.000000  208.000000  208.000000   
mean     0.013420    0.010709    0.010941    0.009290    0.008222    0.007820   
std      0.009634    0.007060    0.007301    0.007088    0.005736    0.005785   
min      0.000800    0.000500    0.001000    0.000600    0.000400    0.000300   
25%      0.007275    0.005075    0.005375    0.004150    0.004400    0.003700   
50%      0.011400    0.009550    0.009300    0.007500    0.006850    0.005950   
75%      0.016725    0.014900    0.014500    0.012100    0.010575    0.010425   
max      0.070900    0.039000    0.035200    0.044700    0.039400    0.035500   

               57          58          59  
count  208.000000  208.000000  208.000000  
mean     0.007949    0.007941    0.006507  
std      0.006470    0.006181    0.005031  
min      0.000300    0.000100    0.000600  
25%      0.003600    0.003675    0.003100  
50%      0.005800    0.006400    0.005300  
75%      0.010350    0.010325    0.008525  
max      0.044000    0.036400    0.043900  

[8 rows x 60 columns]

In [ ]:
data[60].value_counts() # Unique value counts in col 60

M    111
R     97
Name: 60, dtype: int64

111 Mine Examples while 97 Rock examples

In [ ]:
#separating features and targets
X=data.drop(columns=60,axis=1)
Y=data[60]

In [ ]:
x_train,x_test,y_train,y_test=train_test_split(X,Y,test_size=0.1,stratify=Y,random_state=1)

In the above example,<br>
<li><b>test_size</b> specifies the percentage of the whole datset that needs to be in the test set.</li>
<li><b>stratify</b> specifies the column based on which the split needs to take place. i.e., the test set will have the same ratio of the labels as the training data</li>
<li><b>random_state</b> specifies a particular order by which the data is split, which helps in code reusability across different platforms.</li>

In [ ]:
print(x_test.shape[0]/X.shape[0]) # check the test_size as specified by us
print(x_train.shape[0]/X.shape[0])

0.10096153846153846
0.8990384615384616


#Model Initialization and Training

In [ ]:
model=LogisticRegression() # because it is a binary classification problem
model.fit(x_train, y_train)


LogisticRegression()

#Model Valuation

In [ ]:
#accuracy on training data
y_train_predict=model.predict(x_train)
print(accuracy_score(y_train,y_train_predict))

0.8342245989304813


In [ ]:
#accuracy on test data
y_test_predict=model.predict(x_test)
print(accuracy_score(y_test,y_test_predict))

0.7619047619047619


So, we got a 83.4 % Accuracy in the training data while an 76.2 % Accuracy in the test data.